src/components/editor/resourceProfileMapping/ResourceProfileMapping.tsx

```diff
@@ -5,42 +5,113 @@ import {
     profile?.snapshot?.element;
-  const structuredefSettings: string = "structuredefSettings" as string;
-
-  const renderTree: JSX.Element[] | undefined = attributes?.map(
-    (attribute: IElementDefinition, index: number) => {
-      return (
-        <div
-          key={index}
-          onClick={() => {
-            dispatch(selectAttributeId(attribute?.id));
-          }}
-        >
-          {attribute.id}
-        </div>
-      );
-    }
+  const structuredefSettings: string = "structuredefSettings";
+  const classes = useStyles();
+
+  let attributesToRender: RenderTree[] = [];
+  if (attributes) {
+    attributes.forEach((attribute) => {
+      if (attribute.path) {
+        attribute.path.split(".").reduce(
+          (render: any, name: string) => {
+            let node = (render.children = render.children || []).find(
+              (item: RenderTree) => item.name === name
+            );
+            if (!node && attribute.path) {
+              render.children.push({ name, id: attribute.path });
+            }
+            return node;
+          },
+          { children: attributesToRender }
+        );
+      }
+    });
+  }
+
+  const treeItemContent = (nodes: RenderTree) => (
+    <span className={classes.treeItem}>
+      {nodes.children ? (
+        <Folder className={classes.iconTreeItem} />
+      ) : (
+        <LocalOffer className={classes.iconTreeItem} />
+      )}
+      {nodes.name}
+    </span>
   );
 
-  return (
-    <div>
-      <div onClick={() => dispatch(selectAttributeId(structuredefSettings))}>
-        Structure definition Settings
-      </div>
-      {renderTree}
-    </div>
+  const renderTree = (nodes: RenderTree) => (
+    <TreeItem
+      key={nodes.id}
+      nodeId={nodes.id}
+      label={treeItemContent(nodes)}
+      onClick={() => dispatch(selectAttributeId(nodes.id))}
+    >
+      {Array.isArray(nodes.children)
+        ? nodes.children.map((node) => renderTree(node))
+        : null}
+    </TreeItem>
   );
+
+  if (attributesToRender && profile) {
+    return (
+      <div>
+        <Typography variant="h1" className={classes.resourceId}>
+          {profile.id}
+        </Typography>
+        <TreeView
+          className={classes.root}
+          defaultCollapseIcon={<ArrowDropDown />}
+          defaultExpandIcon={<ArrowRight />}
+          defaultExpanded={[attributesToRender[0].id]}
+        >
+          <TreeItem
+            nodeId="0"
+            onClick={() => dispatch(selectAttributeId(structuredefSettings))}
+            label={
+              <span
+                className={`${classes.treeItem} ${classes.textTreeItemMeta}`}
+              >
+                <Settings
+                  className={`${classes.iconTreeItem} ${classes.iconTreeItemMeta}`}
+                />
+                Metadatas
+              </span>
+            }
+          />
+          {renderTree(attributesToRender[0])}
+        </TreeView>
+      </div>
+    );
+  }
+
+  return <div>Error</div>;
```

Comment:
I tend to do the opposite: 

if (my error condition):
return <>Error</>

return <>My result component<>

Reply:
I was checking if attributesToRender && profile were existing before rendering to resolve 'undefined' state issues